x-pack/plugins/infra/server/lib/adapters/metrics/adapter_types.ts

```diff
@@ -38,12 +38,13 @@ export enum InfraMetricModelMetricType {
   series_agg = 'series_agg',
   positive_only = 'positive_only',
   derivative = 'derivative',
+  count = 'count',
 }
 
 export interface InfraMetricModel {
   id: string;
   requires: string[];
-  index_pattern: string;
+  index_pattern: string | string[];
```

Comment:
Can you explain what these changes do and why they were necessary?

Reply:
The `index` argument for the Elasticsearch client can be both a string or an array of strings. Originally it only worked with one index pattern but I realized that there are times we need to do 2.